### ISSUE TITLE
[Quest] Bloodmyst Isle (Draenei starter area) broken quests

### ISSUE THREAD

**Issue by zimms** [unified-db/Database#73]
Assuming these are primarily DB issues, and not SD2/ACID.

[9711] Matis the Cruel: He's supposed to "faint" at low health so you can capture him to complete the quest.  Instead, he's fully killable, which makes this quest not completable.  This seems irrespective of using the flare gun to call for help in capturing him.

[9740] The sun gate: The sun gate can't be activated, even after disabling all 4 controllers.  This prevents the player from being able to destroy the sun gate and thus complete the quest.

[9759] Ending their world: Two issues.  This quest has more pre-reqs than it's supposed to (must have completed both "The sun gate" and "Clearing the way", neither of which seem to be pre-reqs on live).  The larger issue preventing completion of this quest is that the NPC that you're supposed to be escorting (Demolitionist Legoso) doesn't do anything (I guess the escort part isn't implemented, although I see his skills are defined in ACID).  Alone, this is only a significant annoyance.  However, without him, he can't detonate the explosives, and thus cannot trigger Sironas to be attackable.  Quest completion is based on defeating Sironas, so this quest isn't completable.

[9762] The unwritten prophecy: The final quest in the long chain.  This isn't completely broken, as it's completable, but is too easy to miss.  On blizzard live servers, after turning in "Ending their world" you get a "victory parade" event where many NPCs show up at Blood Watch, including Velen, who rewards you using this quest.  In current UDB, the "victory parade" doesn't happen, and thus the player must go to Velen in the Exodar to complete it.  This isn't obvious since there's no guidance text to tell the player that Velen is looking for them since he doesn't show up in Blood Watch.

**Comment by Grz3s** [comment]
Thank you for this report...
i'll look at this soon

**Comment by Grz3s** [comment]
small update for this one:
Bad news is:
Draenei starter area - is the worst scripted starting zone ...:( 
Problems starts almost from begining (1st quest)...
- broken quests / wrong relations between them 
- missing spawns / waypoints 
- etc...
Good news is:
It will change soon*

**Comment by Grz3s** [comment]
@Rushor have a question....
Can you pls look at your sources... and check if you have sniff for "Kudrii" c.16742 in Exodar
i cant find spell (id), she casts every min (small event)....
Would be nice to have her fixed... :)
thx

**Comment by Rushor** [comment]
on my way

**Comment by Rushor** [comment]
waypoints for her:

```sql
-- Pathing for  Entry: 16742 'UDB FORMAT' 
SET @GUID := XXXXXX;
UPDATE `creature` SET `position_x`=-3893.313,`position_y`=-11485.78,`position_z`=-135.9678 WHERE `guid`=@GUID;
DELETE FROM `creature_movement` WHERE `id`=@GUID;
INSERT INTO `creature_movement` (`id`,`point`,`position_x`,`position_y`,`position_z`,`waittime`,`script_id`,`textid1`,`textid2`,`textid3`,`textid4`,`textid5`,`emote`,`spell`,`wpguid`,`orientation`,`model1`,`model2`) VALUES
(@GUID,1,-3888.563,-11494.28,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0),
(@GUID,2,-3888.063,-11491.53,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0),
(@GUID,3,-3888.813,-11488.03,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0),
(@GUID,4,-3890.563,-11485.78,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0),
(@GUID,5,-3892.563,-11485.53,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0),
(@GUID,6,-3893.313,-11485.78,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0),
(@GUID,7,-3892.313,-11486.53,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0),
(@GUID,8,-3890.813,-11486.53,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0),
(@GUID,9,-3889.063,-11487.78,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0),
(@GUID,10,-3887.813,-11490.03,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0),
(@GUID,11,-3888.313,-11493.53,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0),
(@GUID,12,-3889.563,-11494.53,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0),
(@GUID,13,-3888.563,-11494.28,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0),
(@GUID,14,-3888.063,-11491.53,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0),
(@GUID,15,-3888.813,-11488.03,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0),
(@GUID,16,-3890.563,-11485.78,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0),
(@GUID,17,-3892.563,-11485.53,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0),
(@GUID,18,-3893.313,-11485.78,-135.9678,0,0,0,0,0,0,0,0,0,0,0,0,0);
-- 0x1C39C04240105980000025000042CCC4 .go -3888.563 -11494.28 -135.9678
```

spellcast:

```sql
ServerToClient: SMSG_SPELL_START (0x03F5) Length: 106 ConnIdx: 1 Time: 11/15/2015 11:15:37.891 Number: 49
(Cast) CasterGUID: Full: 0x1C39C04240105980000025000042CCC4 Creature/0 R3696/S37 Map: 530 Entry: 16742 Low: 4377796
(Cast) CasterUnit: Full: 0x1C39C04240105980000025000042CCC4 Creature/0 R3696/S37 Map: 530 Entry: 16742 Low: 4377796
(Cast) CastID: 0
(Cast) SpellID: 32990 (32990)
```

order:
waypoint one - stop wp - wait 24 seconds - cast spell - wait further 23 seconds - run to waypoint 6
waypoint six - stop wp - oneshot talk emote - wait 17 seconds - run to waypoint 1

**Comment by Rushor** [comment]
```sql
-- 
-- Kudrii SAI
SET @ENTRY := 16742;
UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`=@ENTRY;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,0,0,0,25,0,100,0,0,0,0,0,53,0,16742,1,0,0,0,1,0,0,0,0,0,0,0,"Kudrii - On Reset - Start Waypoint"),
(@ENTRY,0,1,0,40,0,100,0,1,16742,0,0,80,@ENTRY*100+00,2,0,0,0,0,1,0,0,0,0,0,0,0,"Kudrii - On Waypoint 1 Reached - Run Script"),
(@ENTRY,0,2,0,40,0,100,0,6,16742,0,0,80,@ENTRY*100+01,2,0,0,0,0,1,0,0,0,0,0,0,0,"Kudrii - On Waypoint 6 Reached - Run Script");

-- Actionlist SAI
SET @ENTRY := 1674200;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=9;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,9,0,0,0,0,100,0,0,0,0,0,54,47000,0,0,0,0,0,1,0,0,0,0,0,0,0,"On Script - Pause Waypoint"),
(@ENTRY,9,1,0,0,0,100,0,24000,24000,0,0,11,32990,0,0,0,0,0,1,0,0,0,0,0,0,0,"On Script - Cast 'Enchanting Cast Visual'");

-- Actionlist SAI
SET @ENTRY := 1674201;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=9;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@ENTRY,9,0,0,0,0,100,0,0,0,0,0,54,17000,0,0,0,0,0,1,0,0,0,0,0,0,0,"On Script - Pause Waypoint"),
(@ENTRY,9,1,0,0,0,100,0,1000,1000,0,0,66,0,0,0,0,0,0,8,0,0,0,0,0,0,4.934527,"On Script - Set Orientation 4,934527"),
(@ENTRY,9,2,0,0,0,100,0,1000,1000,0,0,5,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"On Script - Play Emote 1");

SET @PATH := 16742;
DELETE FROM `waypoints` WHERE `entry`=16742;
INSERT INTO `waypoints` (`entry`, `pointid`, `position_x`, `position_y`, `position_z`, `point_comment`) VALUES 
(@PATH,1,-3888.563,-11494.28,-135.9678,'Kudrii'),
(@PATH,2,-3888.063,-11491.53,-135.9678,'Kudrii'),
(@PATH,3,-3888.813,-11488.03,-135.9678,'Kudrii'),
(@PATH,4,-3890.563,-11485.78,-135.9678,'Kudrii'),
(@PATH,5,-3892.563,-11485.53,-135.9678,'Kudrii'),
(@PATH,6,-3893.313,-11485.78,-135.9678,'Kudrii'),
(@PATH,7,-3892.313,-11486.53,-135.9678,'Kudrii'),
(@PATH,8,-3890.813,-11486.53,-135.9678,'Kudrii'),
(@PATH,9,-3889.063,-11487.78,-135.9678,'Kudrii'),
(@PATH,10,-3887.813,-11490.03,-135.9678,'Kudrii'),
(@PATH,11,-3888.313,-11493.53,-135.9678,'Kudrii');
```

i think you can convert this :)

**Comment by Grz3s** [comment]
lol :) ...  thank you very much :)

**Comment by Rushor** [comment]
let me know if you need more

**Comment by Grz3s** [comment]
Hey @Rushor 
Can you pls check your sources... if there is any info about elementals that are spawn in "The Crystal Hall" in Exodar.. i cant get any info ... not even id's :/
Players can see them after using "Elemental Sapta" from Seer Skaltesh
They have non-selectable flag + invisible aura...
thank you

**Comment by Rushor** [comment]
>>Hey @Rushor Can you pls check your sources... if there is any info about elementals that are spawn >>in "The Crystal Hall" in Exodar.. i cant get an…

datas still needed? i just saw the comment :D

**Comment by Grz3s** [comment]
i have found all of them....
```quote
18823 -- Minor Water Spirit
18824 -- Minor Fire Spirit
18825 -- Minor Air Spirit
18826 -- Minor Earth Spirit
```
and now im trying to find what aura they have.... (invisibility)
i did fresh sniff... (becouse of nwe wow patch) ... parser doesnt work :/ .. i cant check it...
So if you have that info it would be nice....
Also any info about them would be nice :)
- unitflags ... movements,,, possitions etc..
thx

**Comment by Grz3s** [comment]
(...)
[9740] The sun gate: The sun gate can't be activated, even after disabling all 4 controllers. This prevents the player from being able to destroy the sun gate and thus complete the quest.
(...)

It is impossible to fix this quest using DB tools...
Our main Object = 182026 Sun Gate ... has Flags = 20
•4 = Untargetable
•16 = Player cant interact with the object. <--- this flag makes problems for us...
Flag 16 is removed from this object .. after player "uses" 4 other objects. (Sunhawk Portal Controler)

@xfurry -- can sd2 script temp. change flag in object? (can also this be controled after x numer objects used?)

**Comment by xfurry** [comment]
@Grz3s I need take a look into this in detail, but it should be doable.

**Comment by Grz3s** [comment]
i can give you vid... and sniff (from WoD) if you need it.

**Comment by xfurry** [comment]
Yes, a vid would help.

**Comment by Grz3s** [comment]
https://www.youtube.com/watch?v=yXVi9jBOQq4&feature=youtu.be

visual spells... may look like they comming from objects(crystals) but they not :) ...
we have invis. npcs inside objects...and they casts spells...  (i've got this part done already)
Also...  You may find server side spell in these 4 crystals....
```code
object 184850  has spell = 38096
```
I think its used to kill these hidden npcs ... so visual spell will disappear (maybe  im wrong)
But from sniff... i see that after object ussage - spell "5" is casted on invis. npcs ;)
Unfortunately sniff says NULL about 38096 (maybe parser is not good enough) :)
Anyway i did that trick using db_scripts_on_go_use...

Also:
This maybe not that important... but i noticed ... when player uses that gameobject... i mean both of them...  crystals and sun gate ...
Its done by this spell:
ID - 8386 Attacking
I dont know how our system "use_object" works ... but maybe this is something to look at ?

BTW:
You can implement working part with visuals ...
http://paste2.org/XUe6ZHJe

Thank you for any help with this one;)

**Comment by xfurry** [comment]
Here is an idea on how we can do this without SD2.

```sql
DELETE FROM dbscripts_on_creature_death WHERE id IN (17886);
INSERT INTO dbscripts_on_creature_death (id, delay, command, datalong, datalong2, buddy_entry, search_radius, data_flags, dataint, dataint2, dataint3, dataint4, x, y, z, o, comments) VALUES
(17886, 1, 31, 17886, 80, 0, 0, 8, 0, 0, 0, 0, 0, 0, 0, 0, '[DND]Sunhawk Portal Controller - terminate script if alive'),
(17886, 5, 27, 8, 0, 182026, 60, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Sun Gate - allow GO interact');
```

**Comment by Grz3s** [comment]
:+1:

**Comment by Grz3s** [comment]
```quote
[9759] Ending their world: Two issues. This quest has more pre-reqs than it's supposed to (must have completed both "The sun gate" and "Clearing the way", neither of which seem to be pre-reqs on live).
```
The Sun Gate quest... must be done before 9759 becomes available...
Clearing the way - on live can be done in same time with 9759
I'll correct that.

**Comment by Grz3s** [comment]
hey @xfurry 
I started collecting all 'needed' data for ESCORT quest
https://www.youtube.com/watch?v=qvh5fTDfjCY
```quote
[9759] Ending their world
```
-- i know you like escort quests need to be done by sd2
I can say .. i got all we need....but.. i need your help with one npc that is a key in this quest and i would like to have him fixed before escort.

This npc:
http://www.wowhead.com/npc=17678/sironas#screenshots
Pls look at pictures... She recives 4 channel spells from another hidden npc's....
this spell:
```code
ID - 31611 Bloodmyst Tesla
Effect 0: Id 6 (SPELL_EFFECT_APPLY_AURA)
BasePoints = 1
Targets (38, 0) (TARGET_SCRIPT, NO_TARGET)
Aura Id 23 (SPELL_AURA_PERIODIC_TRIGGER_SPELL), value = 1, misc = 0 (0), miscB = 0, periodic = 1000
```
I added this spell for these hidden npcs:
```sql
-- Bloodmyst Tesla Coil
Delete from `creature_ai_scripts` where `creature_id`= 17979;
Insert into `creature_ai_scripts` values
('1797901','17979','1','0','100','1','0','0','0','0','11','31611','0','6','0','0','0','0','0','0','0','0','Bloodmyst Tesla Coil - Cast Channel Earth Shield while OOC');
UPDATE creature_template SET AIName='EventAI' WHERE `entry` = '17979';
```
spell runs fine: ... but im reciving error in console:
```code
ERROR:Aura::TriggerSpell: Spell 31611 have 0 in EffectTriggered[0], not handled custom case?
```

Ok so i tried to fix this:
First i removed spell from her template_addon = 31612 -- this spell should be effect of 31611.
Then i found this line in SpellAuras.ccp
```code
//                    // Bloodmyst Tesla
//                    case 31611: break;
``` 
so i added this:
```code
                    case 31611: 
                        trigger_spell_id = 31612;
                        break;
```
now ... i can see that she recives that second spell... but now i have another error in console:
```sql
ERROR:SpellAuraHolder (Spell 31611) is in process but attempt removed at SpellAuraHolder (Spell 31612) adding, need add stack rule for Unit::RemoveNoStackAurasDueToAuraHolder
```
?? because 4 npcs.. casting same spell on her???
And here i stopped....
Can you pls help me with it?
Anyway... this spells and aura ... will be removed during escort quest// (look at 4:40 vid above)

Once we'll finish with this... i'll give you all data for that escort .. ;)
Thanks

**Comment by xfurry** [comment]
Thanks,
I will check this soonish.

**Comment by Grz3s** [comment]
here her Aura update if its needed:
```code
ServerToClient: SMSG_AURA_UPDATE (0x2B9F) Length: 211 ConnIdx: 1  18:19:56.788 Number: 3755
UpdateAll: True
UnitGUID: Full: 0x2009E442401143800000250000000A8E  Map: 530 Entry: 17678 Low: 2702
AurasCount: 5
[0] Slot: 0
[0] HasAura: True
[0] SpellID: 31612 (31612)
[0] SpellXSpellVisualID: 10946
[0] Flags: NoCaster, Positive (3)
[0] ActiveFlags: 1
[0] CastLevel: 20
[0] Applications: 0
[0] Int56 Count: 0
[0] Effect Count: 0
[0] HasCastUnit: False
[0] HasDuration: False
[0] HasRemaining: False
[1] Slot: 1
[1] HasAura: True
[1] SpellID: 31611 (31611)
[1] SpellXSpellVisualID: 10945
[1] Flags: Positive (2)
[1] ActiveFlags: 1
[1] CastLevel: 1
[1] Applications: 0
[1] Int56 Count: 0
[1] Effect Count: 0
[1] HasCastUnit: True
[1] HasDuration: False
[1] HasRemaining: False
[1] CastUnit: Full: 0x2009E44240118EC000002500017B35C1  Map: 530 Entry: 17979 Low: 24851905
[2] Slot: 2
[2] HasAura: True
[2] SpellID: 31611 (31611)
[2] SpellXSpellVisualID: 10945
[2] Flags: Positive (2)
[2] ActiveFlags: 1
[2] CastLevel: 1
[2] Applications: 0
[2] Int56 Count: 0
[2] Effect Count: 0
[2] HasCastUnit: True
[2] HasDuration: False
[2] HasRemaining: False
[2] CastUnit: Full: 0x2009E44240118EC000002500007B35C1  Map: 530 Entry: 17979 Low: 8074689
[3] Slot: 3
[3] HasAura: True
[3] SpellID: 31611 (31611)
[3] SpellXSpellVisualID: 10945
[3] Flags: Positive (2)
[3] ActiveFlags: 1
[3] CastLevel: 1
[3] Applications: 0
[3] Int56 Count: 0
[3] Effect Count: 0
[3] HasCastUnit: True
[3] HasDuration: False
[3] HasRemaining: False
[3] CastUnit: Full: 0x2009E44240118EC00000250001FB35C1  Map: 530 Entry: 17979 Low: 33240513
[4] Slot: 4
[4] HasAura: True
[4] SpellID: 31611 (31611)
[4] SpellXSpellVisualID: 10945
[4] Flags: Positive (2)
[4] ActiveFlags: 1
[4] CastLevel: 1
[4] Applications: 0
[4] Int56 Count: 0
[4] Effect Count: 0
[4] HasCastUnit: True
[4] HasDuration: False
[4] HasRemaining: False
[4] CastUnit: Full: 0x2009E44240118EC00000250000FB35C1  Map: 530 Entry: 17979 Low: 16463297

**Comment by xfurry** [comment]
@Grz3s these channeling auras can be fixed with something like this:
I will push some core fix, to avoid errors for this spell, as there is no usage of the periodic dummy aura.

However, please don't push the code to acid yet, as I might add more to this, once I script the escort part.

```sql
DELETE FROM spell_script_target WHERE entry IN (31611);
INSERT INTO spell_script_target (entry,type,targetEntry,inverseEffectMask) VALUES
(31611,1,17678,0);

UPDATE creature_template SET AIName='EventAI' WHERE entry=17979;
DELETE FROM creature_ai_scripts WHERE creature_id=17979;
INSERT INTO creature_ai_scripts VALUES 
('1797901','17979','30','0','100','1','5','17678','0','0','11','31611','0','32','0','0','0','0','0','0','0','0','Bloodmyst Tesla Coil - Cast Channel Earth Shield on Received AI event');


-- Creature id: 17678
UPDATE creature_template SET AIName='EventAI' WHERE entry=17678;
DELETE FROM creature_ai_scripts WHERE creature_id=17678;
INSERT INTO creature_ai_scripts VALUES 
('1767801','17678','9','0','100','1','0','30','9000','14000','11','12742','4','32','0','0','0','0','0','0','0','0','Sironas - Cast Immolate'),
('1767802','17678','9','0','100','1','0','5','11000','15000','11','10966','1','1','0','0','0','0','0','0','0','0','Sironas - Cast Uppercut'),
('1767803','17678','0','0','100','1','4000','7000','16000','21000','11','8282','1','33','0','0','0','0','0','0','0','0','Sironas - Cast Curse of Blood'),
('1767804','17678','0','0','100','1','8000','10000','15000','19000','11','13338','4','1','0','0','0','0','0','0','0','0','Sironas - Cast Curse of Tongues'),
('1767805','17678','1','0','100','0','1000','1000','0','0','45','5','70','0','0','0','0','0','0','0','0','0','Sironas - Send Custom event on after Spawn'),
('1767806','17678','21','0','100','0','0','0','0','0','45','5','70','0','0','0','0','0','0','0','0','0','Sironas - Send Custom event on Reached Home');
```

**Comment by Grz3s** [comment]
All guids you need ...
'Draenei Explosives' - desp time set to "-300"   -  they must be despawned by your script ...so 5 min delay should be enough .... :)

**Comment by Grz3s** [comment]
It looks like we finished ... this zone  :)
Thx to all who helped here.

**Comment by xfurry** [comment]
@Grz3s did you also update the Wyrmscar Island?

**Comment by Grz3s** [comment]
Yes..
BloodCurse Isle - also.

Draenei Zone is completed .;)
